### DISCUSSION TITLE
Theia Unit and Integration Testing

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#11105] [Q&A] [answered]
Hi,
I'm new to Unit and Integration Testing. I wish to know how theia performs Unit and Integeration Testing, and the packages that are used. Any good component to look into that would help me start ? I am aiming to write Unit and Integration Test for our Custom Components. 

Thanks

**Comment by vince-fugnitto** [comment, marked as answer]
@maheshrajrp thank you for the discussion, we have the following:

- unit tests - using [`mocha`](https://mochajs.org/)
- [api integration tests](https://github.com/eclipse-theia/theia/blob/master/doc/api-testing.md)
- [playwright ui testing](https://github.com/eclipse-theia/theia/tree/master/examples/playwright)

An example of a unit test is the following:

- https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/uri.spec.ts

For unit-testing you can use whichever framework you prefer however :)

**Reply by maheshrajrp** [reply]
Thanks,  @vince-fugnitto . This helps.
